### PR TITLE
Adding a more deterministic matching of command line options.

### PR DESCRIPTION
**Pull request by miskoSplunk**

### CODE REVIEW

File: detections/endpoint/malicious_powershell_executed_as_a_service.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ description: This detection is to identify the abuse the Windows SC.exe to execu
   malicious commands or payloads via PowerShell.
 search: ' `wineventlog_system` EventCode=7045 | eval l_Service_File_Name=lower(Service_File_Name)
   | regex l_Service_File_Name="powershell[.\s]|powershell_ise[.\s]|pwsh[.\s]|psexec[.\s]"
-  | regex l_Service_File_Name="-nop[rofile]*|-w[indowstyle]*\s+hid[den]*|-noe[xit]*|-enc[odedcommand]*"
+  | regex l_Service_File_Name="-nop[rofile\s]+|-w[indowstyle]*\s+hid[den]*|-noe[xit\s]+|-enc[odedcommand\s]+"
```
Comment (MHaggis) [inline detections/endpoint/malicious_powershell_executed_as_a_service.yml:13]: @miskoSplunk Random question, would it make any difference to use eval statements over regex (for performance)? 
```
| eval NoProf=if(match(lower(CommandLine),"-nop"),1,0)
```

```
| eval Hidden=if(match(lower(CommandLine)," hidden") OR match(lower(CommandLine),"-windowstyle") OR match(lower(CommandLine), "-nonewwindow"),1,0)
```
```
| eval Encode=if(match(lower(CommandLine),"[A-Za-z0-9+\/]{44,}([A-Za-z0-9+\/]{4}|[A-Za-z0-9+\/]{3}=|[A-Za-z0-9+\/]{2}==)") OR match(lower(CommandLine),"frombase64string"),5,0)
```

Comment (miskoSplunk) [reply detections/endpoint/malicious_powershell_executed_as_a_service.yml:13]: @MHaggis Apologies for the late reply. 
I'm not sure what would work faster, but I also think we're fine with performance - event 7045 is not at all frequent, so this won't go into regexes section much :)

**Comment by josehelps** [comment]
@miskoSplunk this PR is missing a [test](https://github.com/splunk/security_content/tree/develop/automated_detection_testing) since we are changing around the search. Please submit one and once it passes I believe we can merge this one in, thank you!

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Malicious Powershell Executed As A Service :white_check_mark:
- diskUsage: 139264
- runDuration: 0.681s
- scanCount: 21

**Review by jzsplunk — APPROVED** [review]
regex looks good

**Pull request merged**